Products/Zing/datamaps.py

```diff
@@ -243,6 +238,42 @@ def fact_from_object_map(self, om, parent_device=None, relationship=None, contex
                 f.metadata[ZFact.FactKeys.PLUGIN_KEY] = f.metadata[ZFact.FactKeys.META_TYPE_KEY]
         return f
 
+    def fact_from_incremental_map(self, idm, context=None):
+        f = ZFact.Fact()
+        valid_types = (
+            str, int, long, float, bool, list, tuple, MultiArgs, set
+        )
+
+        objectmap = {k: v for k, v in idm.iteritems()}
+        for k, v in objectmap.items():
+            # These types are currently all that the model ingest service can handle.
+            if not isinstance(v, valid_types):
+                del objectmap[k]
+            elif isinstance(v, MultiArgs):
+                objectmap[k] = v.args
+
+        objectmap['id'] = idm.id
+        f.update(objectmap)
+
+        f.metadata["relationship"] = idm.relname
+        f.metadata[ZFact.FactKeys.PLUGIN_KEY] = idm.plugin_name
+        try:
+            f.metadata["parent"] = idm.parent.getUUID()
+        except Exception:
+            log.debug('parent UUID not found')
+
+        # Hack in whatever extra stuff we need.
+        om_context = (context or {}).get(idm)
+        if om_context is not None:
+            self.apply_extra_fields(om_context, f)
+
+        # FIXME temp solution until we are sure all zenpacks send the plugin
```

Comment:
Still necessary?

Reply:
maybe?  this code does not have full test coverage yet, so i'm leaving it as-is